transport/pool.go

```diff
@@ -119,12 +121,25 @@ type PoolRefiller struct {
 	active int
 }
 
-func (r *PoolRefiller) init(addr string) error {
+func (r *PoolRefiller) init(addr, ks string) error {
 	conn, err := OpenConn(withDefaultPort(addr), nil, r.cfg)
 	if err != nil {
 		return err
 	}
 
+	if ks != "" {
+		if err := validateKeyspace(ks); err != nil {
+			return err
+		}
+		if err := conn.UseKeyspace(ks); err != nil {
+			log.Printf("failed to connect to the keyspace: %s", err)
+			conn.Close()
+			if !isIOError(err) {
+				return fmt.Errorf("keyspace request failed %w", err)
```

Comment:
Should we fail initialization now also when IO error happens?

Reply:
We fail on any error.

We would handle that later with retries - no need to worry now.